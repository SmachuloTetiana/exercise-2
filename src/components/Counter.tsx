import React from "react";
import "./style.css";
import counter from "../store/counter";
import {observer} from "mobx-react";

const Counter: React.FC = () => (
    <main className="container">
        <h1 className="title">Clicks: {counter.count}</h1>

        <div className="button-group">
            <button onClick={() => counter.increment()} className="button-group__action">
                Increment
            </button>
            <button onClick={() => counter.decrement()} className="button-group__action">
                Decrement
            </button>
        </div>
    </main>
);

const CounterObserver = observer(Counter);

export default CounterObserver;
