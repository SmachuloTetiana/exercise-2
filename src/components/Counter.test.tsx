import React from "react";
import {shallow} from "enzyme";
import Counter from "./Counter";

describe("Counter click", () => {
    test("should render", () => {
        const component = shallow(<Counter />);

        expect(component).toMatchSnapshot();
    });
});
