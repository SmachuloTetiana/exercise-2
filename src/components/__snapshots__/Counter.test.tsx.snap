// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter click should render 1`] = `ShallowWrapper {}`;
